In [3]:
import numpy as np
from scipy import stats

1. Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трех группах случайно выбранных спортсменов: Футболисты: 173, 175, 180, 178, 177, 185, 183, 182. Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180. Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170.

In [38]:
fp = np.array([173, 175, 180, 178, 177, 185, 183, 182])
hp = np.array([177, 179, 180, 188, 177, 172, 171, 184, 180])
wl = np.array([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170])

Решение с помощью функции:

In [5]:
stats.f_oneway(fp, hp, wl)

F_onewayResult(statistic=5.500053450812596, pvalue=0.010482206918698694)

Решение вручную:

In [19]:
k = 3
n1 = len(fp)
n2 = len(hp)
n3 = len(wl)
n = n1 + n2 + n3

In [20]:
y1_mean = np.mean(fp)
y2_mean = np.mean(hp)
y3_mean = np.mean(wl)
y_mean = sum([y1*n1, y2*n2, y3*n3]) / n
y_mean

176.46428571428572

In [23]:
y = fp + hp + wl
s2 = sum((y - y_mean) ** 2)
s2

830.9642857142854

In [27]:
sf2 = ((y1_mean - y_mean) ** 2) * n1 + ((y2_mean - y_mean) ** 2) * n2 + ((y3_mean - y_mean) ** 2) * n3
sf2

253.9074675324678

In [28]:
so2 = sum((fp - y1_mean) ** 2) + sum((hp - y2_mean) ** 2) + sum((wl - y3_mean) ** 2)
so2

577.0568181818181

In [31]:
sigmaf2 = (1 / (k - 1)) * sf2
sigmaf2

126.9537337662339

In [32]:
sigmao2 = so2 / (n - k)
sigmao2

23.082272727272724

In [37]:
fh = sigmaf2 / sigmao2
fh

5.500053450812599

Табличное значение критерия фишера для a(0.05) = 3.39
Так как 5.5 > 3.39 верна альтернативная гипотеза.
Следовательно существуют различия среднего роста в зависимости от вида спорта, в представленных группах